src/Modules/Common.lua

```diff
@@ -816,3 +817,14 @@ function string:split(sep)
 	return fields
 end
 
+-- Ceil function with optional base parameter
+function ceil(x, base)
+	base = base or 1
+	return base * m_ceil(x/base)
+end
+
+-- Ceil function with optional base parameter
+function floor(x, base)
+	base = base or 1
+	return base * m_floor(x/base)
+end
```

Comment:
A function like this already exists it uses number of decimals as the parameter though.
```lua
--- Rounds down a number to the nearest <dec> decimal places
---@param val number
---@param dec number
---@return number
function floor(val, dec)
	if dec then
		local mult = 10 ^ dec
		return m_floor(val * mult + 0.0001) / mult
	else
		return m_floor(val)
	end
end
```

Reply:
Maybe name it `ceil_b` (ceil to base) to distinguish from the version with decimal precision

Reply:
I'm not sure how i feel about adding yet another implementation of floor to the code base. I'd need to convert all other instances to maintain consistency too.

Reply:
Converting the remaining uses would be out of scope imo. But I agree that that 2nd parameter in the new `ceil` function might be mistaken as decimal precision, instead of base. Tho the two are interchangeable since `ceil_b(X, .01) == ceil(X, 2)`. 

Afaik the `ceil(X,base)` function was introduced here, so that should be renamed to `ceil_b(X,base)`.